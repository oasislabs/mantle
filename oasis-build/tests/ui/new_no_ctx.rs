use oasis_std::Context;

#[derive(oasis_std::Service, Default)]
pub struct Counter(u32);

impl Counter {
    pub fn new(ctx: Context) -> Result<Self, String> {
        if true {
            Err("error".to_string())
        } else {
            Ok(Default::default())
        }
    }
}

fn main() {
    oasis_std::service!(Counter);
}
