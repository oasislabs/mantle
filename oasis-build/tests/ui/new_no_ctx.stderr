error: Service constructor must take `&Context` as its first argument.
 --> $DIR/new_no_ctx.rs:7:16
  |
7 |     pub fn new(ctx: Context) -> Result<Self, String> {
  |                ^^^^^^^^^^^^

error: aborting due to previous error

